File: TFG/ftp_tfg/lib/domain/usecases/delete_profile.dart
import '../repositories/ftp_repository.dart';
import '../interfaces/i_delete_profile_use_case.dart';

class DeleteProfile implements IDeleteProfileUseCase {
  final FtpRepository repository;

  DeleteProfile(this.repository);

  @override
  Future<void> execute(int id, String ownerId) =>
      repository.deleteProfile(id, ownerId);
}

File: TFG/ftp_tfg/backend/src/application/use-cases/profiles/delete-profile.ts
import type { ProfileRepository } from '../../../domain/repositories/profile-repository.js';

export class DeleteProfile {
  constructor(private readonly profileRepository: ProfileRepository) {}

  execute(ownerId: string, id: number): Promise<void> {
    return this.profileRepository.delete(ownerId, id);
  }
}

